internal/consumer/consumer.go

```diff
@@ -620,6 +671,26 @@ func (i *InventoryConsumer) RebalanceCallback(consumer *kafka.Consumer, event ka
 		i.Logger.Warnf("consumer rebalance event: %d partition(s) revoked: %v\n",
 			len(ev.Partitions), ev.Partitions)
 
+		// Check if shutdown is already in progress to avoid double commits
+		i.offsetMutex.Lock()
+		shutdownInProgress := i.shutdownInProgress
+		hasOffsets := len(i.OffsetStorage) > 0
+		i.offsetMutex.Unlock()
+
+		if shutdownInProgress {
+			i.Logger.Info("shutdown in progress, skipping rebalance offset commit")
+			i.lockToken = ""
+			i.lockId = ""
```

Comment:
Should we move the resetting of these vars to the shutdown func? It seems "smelly" to reset `lockToken` and `lockId` as a side effect of a rebalance if a shutdown is in progress. It's not immediately apparent to me if we _need_ to reset these here.